libraries/HID/src/HID.cpp

```diff
@@ -133,25 +133,48 @@ bool HID_::setup(USBSetup& setup)
 		}
 		if (request == HID_SET_REPORT)
 		{
-			//uint8_t reportID = setup.wValueL;
-			//uint16_t length = setup.wLength;
-			//uint8_t data[length];
+			uint8_t reportID = setup.wValueL;
+			uint16_t length = setup.wLength;
+			uint8_t data[length];
 			// Make sure to not read more data than USB_EP_SIZE.
 			// You can read multiple times through a loop.
 			// The first byte (may!) contain the reportID on a multreport.
 			//USB_RecvControl(data, length);
+			if ((reportID == 2) && (length == 2))
+			{
+				if (2 == USB_RecvControl(data, length)) 
+				{
+					_keyboardLedsStatus = data[1];
+					if (keyboardLedsStatusReportCallback != 0)
+					{
+						keyboardLedsStatusReportCallback();
+					}
+					return true;
+				}
+			}
 		}
 	}
 
 	return false;
 }
 
+uint8_t HID_::getKeyboardLedsStatus(void)
+{
+	return _keyboardLedsStatus;
+}
+
+void HID_::setKeyboardLedsStatusReportCallback(void (*callback) (void) )
+{
+	keyboardLedsStatusReportCallback = callback;
+}
+
 HID_::HID_(void) : PluggableUSBModule(1, 1, epType),
                    rootNode(NULL), descriptorSize(0),
                    protocol(HID_REPORT_PROTOCOL), idle(1)
 {
 	epType[0] = EP_TYPE_INTERRUPT_IN;
 	PluggableUSB().plug(this);
+  setKeyboardLedsStatusReportCallback( 0 );
```

Comment:
I'd replace `0` with `NULL` to make the purpose clearer. Also, please use the same indentation as the rest of the file.

Reply:
Agreed